sklearn/linear_model/omp.py

```diff
@@ -805,7 +805,7 @@ def fit(self, X, y):
         self : object
             returns an instance of self.
         """
-        X, y = check_X_y(X, y)
+        X, y = check_X_y(X, y, dtype=np.float)
```

Comment:
same remark here: `np.float64` to be explicit. Or add `X = as_float_array(X, copy=False)` after the call to `check_X_y` if both `np.float32` and `np.float64` are supported.